### ISSUE TITLE
nnUNet_train 2d - Performance issue with overloading the main memory usage

### ISSUE THREAD

**Issue by jagh** [MIC-DKFZ/nnUNet#744]
Hi Fabian, I kindly report a performance issue with overloading the main memory usage `~128 GB` when training a 2D U-Net with a custom 2D dataset for task115. 

Which occupies `~128 GB` of main memory and takes `~14 minutes` per epoch when `12 threads` are set in `default_data_augmentation.py`. So if I set it to `4 threads` the main memory usage is reduced to `~48 GB`, but the execution time increases to about `33 minutes` per epoch. 

Specifically, the dataset is composed of 2D axial slices and each image is normalized to a shape of `(1, x, y)` according to the guidelines given for using 2D data with nnUNet. However, I do not experience this performance issue when I use the 3D MICCAI dataset and then train a 2D U-Net.
In addition, I have verified that my system generates results similar to those published in the benchmark. [(Pytorch 1.7.1 compiled with cuDNN 8.1.0.77)](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/expected_epoch_times.md)

In this regard, I wonder if you can provide additional pointers to stabilize memory usage and runtime performance.

**Comment by FabianIsensee** [comment]
Do you have the latest pytorch version? If not can you please update and test again?

**Comment by FabianIsensee** [comment]
Also what is your cudnn version?
Mine: 
In [2]: torch.backends.cudnn.version()
Out[2]: 8200
